@here/harp-webpack-utils/scripts/HarpWebpackConfig.js

```diff
@@ -67,26 +85,22 @@ export function addHarpWebpackConfig(config?: Configuration, harpConfig?: HarpWe
         )
     ];
     if (mainEntry !== undefined) {
-        bundles[0] = WebpackMerge.smart(
-            {
-                entry: {
-                    mapview: mainEntry
-                }
-            },
-            bundles[0]
-        );
+        const entry = typeof mainEntry === "string" ? { mapview: mainEntry } : mainEntry;
+        bundles[0] = WebpackMerge.smart({ entry }, bundles[0]);
     }
     if (harpConfig !== undefined && harpConfig.decoderEntry !== undefined) {
-        const decoderConfig = harpConfig.decoderEntry.endsWith(".ts")
+        const decoderConfig = isTypescript(harpConfig.decoderEntry)
```

Comment:
General, `typescript` should be also part of `HarpWebpackConfig` interface, so we not only rely on automagic detection.